src/lib/optimizer/strategy/subquery_to_join_rule.cpp

```diff
@@ -223,10 +223,76 @@ std::optional<SubqueryToJoinRule::PredicateNodeInfo> SubqueryToJoinRule::is_pred
       return std::nullopt;
     }
 
-  } else if (const auto binary_predicate =
-                 std::dynamic_pointer_cast<BinaryPredicateExpression>(predicate_node.predicate())) {
+  } else if (auto binary_predicate = std::dynamic_pointer_cast<BinaryPredicateExpression>(predicate_node.predicate())) {
     result.join_mode = JoinMode::Semi;
 
+    /**
+     * Identify a subquery in an arithmetic expression and push the arithmetics into the subquery.
+     * e.g. SELECT * FROM a WHERE a.a > 3 * (SELECT SUM(b.a) FROM b WHERE b.b = a.b)
+     * becomes SELECT * FROM a WHERE a.a > (SELECT 3 * SUM(b.a) FROM b WHERE b.b = a.b)
+     * There are 4 cases that are essentially the same: The subquery could be in the left/right operand of the binary
+     * predicate and then it could be the left/right operand of that arithmetic expression.
+     */
+    if (const auto right_arithmetic_expression =
```

Comment:
This should be easy to de-duplicate if instead of left/right_operand, you use arguments.

Reply:
I moved the duplicate code into a helper function.

Reply:
Much better